tools/run_tests/python_utils/report_utils.py

```diff
@@ -46,8 +46,22 @@ def _filter_msg(msg, output_format):
 def render_junit_xml_report(resultset, xml_report, suite_package='grpc',
                             suite_name='tests'):
   """Generate JUnit-like XML report."""
-  root = ET.Element('testsuites')
-  testsuite = ET.SubElement(root, 'testsuite', id='1', package=suite_package,
+  tree = add_junit_xml_results(resultset, suite_package, suite_name, '1')
+  create_xml_report_file(xml_report, tree)
+
+def create_xml_report_file(xml_report, tree):
+  """Generate JUnit-like report file from xml tree ."""
+  # ensure the report directory exists
+  report_dir = os.path.dirname(os.path.abspath(xml_report))
+  if not os.path.exists(report_dir):
+    os.makedirs(report_dir)
+  tree.write(xml_report, encoding='UTF-8')
+
+def add_junit_xml_results(resultset, suite_package, suite_name, id,
```

Comment:
I think this method is a bit ill-defined. If "old_tree" is set (not the best name of the param IMHO), the parameters "suite_package", "suite_name" and "id" are ignored.
Having said that and taking into account that both kokoro and jenkins have no problem processing multiple XML reports in  single run, how about reverting changes to report_utils.py and just generating an XML report for each configuration?